### DISCUSSION TITLE
why the sys_configs has more than one POSCAR

### DISCUSSION THREAD

**Discussion by l98y3j26** [deepmodeling/dpgen#509] [Q&A]
Dear dpgen developers,

I know that the the keyword 'sys_configs' contains structure for the inital configuration for MD simulation in LAMMPS, but why we need more than one inital configurations for MD simulation? What is the difference between these different structures in sys_configs? How to choose the inital configuration for MD simulation without using init_bulk function? what is the relationship between the number of  the the inital configurations for MD simulation and the iteration times.

**Comment by tuoping** [comment]
"sys_configs" provides the initial structures for model_devi. Choose different structures, e. g. fcc, bcc, etc, and several random sets for each kind of structures.
You can certainly generate initial structures manually, instead of using "init_bulk".
The iterations are set by keyword "model_devi_jobs". E.g.
```json
    "model_devi_jobs": [
        { "sys_idx":[0, 1, 2], "temps": [50,100,200], "press":  [1e0, 1e2, 1e4], "traj_freq": 20, "nsteps": 1000,  "ens": "npt-tri", "_idx":0}
        ]
```
The "sys_idx" corresponds to the configurations in "sys_configs". MD with temperature 50K, 100K, 200K, pressure 1e0 bar, 1e2 bar, 1e4 bar, with initial configuration 0, 1, 2 are each run seperately. So in total 27 cases of MD are run.
For details, refer to https://github.com/deepmodeling/dpgen. Welcome to propose changes to README by issues or pulling request for changes. 
PS. This discussion is related to dpgen, not deepmd-kit.

**Reply by l98y3j26** [reply]
Thank you!
1.  I am still confused that in the tutorial of methane, the intial configures are so many (in total 20 inital)  configures for model deviation progess, like below:
  "sys_configs": [
        [
            "/public/data/deepmd-tutorial/data/CH4.POSCAR.01x01x01/01.scale_pert/sys-0004-0001/scale-1.000/00000*/POSCAR"
        ], 
        [
            "/public/data/deepmd-tutorial/data/CH4.POSCAR.01x01x01/01.scale_pert/sys-0004-0001/scale-1.000/00001*/POSCAR"
        ]
    ], 
What is the difference these 20 intial POSCAR? what is the purpose to use so many initial POSCAR which increases the iteration rapidly and increase the cost in time and money.

2. if I want to use the potential to get a very long time,such as 100 ns, simulation, as you said, I need to run 27 cases of 100 ns simulation MD. I think it is very time and money consuming! And if I want to get the potential in different temperature, should I need to run AIMD under different tempreture firstly?

**Reply by AnguseZhang** [reply]
1. About differences about POSCARS. They are generated by perturbing original relaxed structures, and can serve as initial structures for MD. Details can be found in DOI: 10.1103/PhysRevMaterials.3.023804 Figure 1.
2. "what is the purpose to use so many initial POSCAR which increases the iteration rapidly and increase the cost in time and money."  " I think it is very time and money consuming! "  Don't judge arbitrarily. Experiences derive from practices. Methane is just an example. Of course you can decrease the number of structures if there are many components to be explored, such as alloys. As for the 100ns simulation, since you want to get a potential at certain temperature and pressures , how can you obtain such a model but never run 100ns in the same condition? (3x3 temperature and pressure conditions). Mean while, the implementation of DP-GEN has ensured the minimal cost of  data generation, since the exploration is based on DPMD, not AIMD. The former is faster than the latter in several magnitudes.  If you have better and easier ways to prepare data and ensure the quality of the model trained from your data, you're welcomed to propose and implement your method in DP-GEN.

**Reply by l98y3j26** [reply]
Thanks so much! and I still have some questions.
1.  Although I know that it is better to run AIMD as long as possible to get a good initial data for training, in general, how long should I run AIMD as the intial data to train the potential. ?  
2.  why we need to perturb the  original relaxed structures intially? 
3.  Lets take a example. I have run an AIMD for one silicate cluster in the center of a 100 water box at 300K. After training, I can got 4 potential models. If I want to see ten silicate clusters in the ceter of the 1000 water box still at 300K, what should I do? should I make one new structure or more than one new structures to do the model deviation? I think it is can not use init_bulk function.
4. Is there a Wechat group for DPMD? Maybe it is more convenious for people to communicate confusions.

**Reply by AnguseZhang** [reply]
1. No general rule. AIMD just serves to prepare a rough initial data set for DP-GEN,  so it won't need too much data, such as several hundred steps each trajectory (also depends on systems). You can also refer to some related papers. 
2. If you use only one structure and generate only one trajectory, the structures sampled in this MD trajectory is expected to be highly correlated with each other, and hence the quality of training data may be low. As a consequence, we perturb the relaxed structures and generate several independent trajectories. 
3. It's not necessary for you to use `init_bulk` to prepare initial data. If you have your own methods, just use them. 
4. Yes, there is WeChat group for Deep Potential, you can contact the admin whose WeChat ID is deeppotential, and let him invite you in the group. You can talk to other users in the group. However, DP developers don't have the responsibility to answer the questions in WeChat group, and the only official ways to solve the problem is github/gitee's issues or discussions.

Yuzhi
